controller/backup_controller.go

```diff
@@ -434,99 +510,139 @@ func (bc *BackupController) backupCreation(log logrus.FieldLogger, engineClient
 		biChecksum = bi.Status.Checksum
 	}
 
-	backup.Status.State = types.BackupStateInProgress
-	event(nil, backup.Status.State, backup, volume)
+	var state types.BackupState
+	_, err = engineClient.SnapshotBackup(backup.Name, backup.Spec.SnapshotName, url, biName, biChecksum, backup.Spec.Labels, credential)
```

Comment:
No. It'll do the same operation again.

If this operation runs twice and since the input parameters are the same, so in the end, it won't create two backups.